src/components/QrScan/QrScan.js

```diff
@@ -158,28 +158,62 @@ const QrScan = () => {
     const handleScan = (data) => {
       if (healthCardPattern.test(data)) {
         const match = data.match(healthCardPattern);
-        if (match.groups.multipleChunks) {
-          const chunkCount = +match.groups.chunkCount;
-          const currentChunkIndex = +match.groups.chunkIndex;
-          let tempScannedCodes = [...scannedCodesRef.current];
-          if (tempScannedCodes.length !== chunkCount) {
-            tempScannedCodes = new Array(chunkCount);
-            tempScannedCodes.fill(null, 0, chunkCount);
-          }
+        if (!multipleHC) {
```

Comment:
I think this needs to be refactored. The scanner shouldn't really change at all except appending the latest scanned codes to the running list in context. Maybe the state variables in context need to be changed to make that work.